src/browser/pages/vscode.ts

```diff
@@ -155,14 +203,50 @@ export function setBodyBackgroundToThemeBackgroundColor(document: Document, loca
     )
   }
 
-  document.body.style.background = editorBgColor
+  _document.body.style.background = editorBgColor
 
   return null
 }
 
+/**
+ * A helper function to encapsulate all the
+ * logic used in this file.
+ *
+ * We purposely include all of this in a single function
+ * so that it's easier to test.
+ */
+export function main(_document: Document | undefined, _window: Window | undefined, _localStorage: Storage | undefined) {
+  const errorMsgPrefix = "[vscode]"
+
+  if (!_document) {
+    throw new Error(`${errorMsgPrefix} Could not run code-server vscode logic. document is undefined.`)
+  }
+
+  if (!_window) {
+    throw new Error(`${errorMsgPrefix} Could not run code-server vscode logic. window is undefined.`)
+  }
+
+  if (!_localStorage) {
+    throw new Error(`${errorMsgPrefix} Could not run code-server vscode logic. localStorage is undefined.`)
```

Comment:
IMO errors should be purely the error since that reduces boilerplate a bit and makes them more flexible. (`document is undefined` for example.)

Then in the catch where we log the error:
```javascript
catch (error) {
  console.error("[vscode] failed to initialize VS Code")
  console.error(error.message) // Or console.log(error) to print a stack trace
}
```

Reply:
> IMO errors should be purely the error since that reduces boilerplate a bit and makes them more flexible. (document is undefined for example.)

Great point! I'm on board with that. I'll refactor!